adr/0025-composition.md

```diff
@@ -0,0 +1,282 @@
+3. `images` (~~`[]string`~~ `[]{ name, newName?, newTag? }`): adopt
+   [`ImageTagTransformer` semantics from Kustomize](https://kubectl.docs.kubernetes.io/references/kustomize/builtins/#_imagetagtransformer_)
+   - `name`: the image name present in the component manifests
+   - `newName` (optional): the new image name you wish to use
+     (useful for changing registry locations)
+   - `newTag` (optional): the new image tag you wish to reference
+     (useful for updating version tags without modifying or relying on Helm chart values)
+
+```yaml
+kind: ZarfPackageConfig
+metadata:
+  name: mattermost
+  description: "UDS Mattermost Package"
+  version: "9.9.0-uds.0"
+
+variables:
+  - name: SUBDOMAIN
+    description: "Subdomain for Mattermost"
+    default: "chat"
+  - name: DOMAIN
+    default: "uds.dev"
+  - name: ACCESS_KEY
+    description: "Access Key for S3 compatible storage"
+  - name: SECRET_KEY
+    description: "Secret Key for S3 compatible storage"
+  - name: DB_PASSWORD
+    description: "Database Password for Mattermost"
+
+components:
+  - name: mattermost
+    description: "Deploy Mattermost"
+    required: true
+
+    ## The `mattermost` component is now a self-contained, deployable unit.
+    ## Thus, all the configuration from `common/zarf.yaml` has been inlined
+    ## into this example. We no longer import a `common` "skeleton package".
+    #
+    # import:
+    #   path: common
+    # only:
+    #   flavor: upstream
+    charts:
+      - name: uds-mattermost-config
+        namespace: mattermost
+        version: 0.1.0
+        localPath: ./chart
+        valuesFiles:
+          - values/config-values.yaml
+      - name: mattermost-enterprise-edition
+        namespace: mattermost
+        url: https://helm.mattermost.com
+        gitPath: chart
+        version: 2.6.55
+        valuesFiles:
+          - values/common-values.yaml
+
+          ## No longer needed because this was only used for replacing
+          ## image tags, see new `images` API usage below.
+          #
+          # - values/upstream-values.yaml
+
+    ## Kustomize-style image replacements facilitate proper composition.
+    ## This makes it easy for downstream components to override image tags
+    ## without knowing anything about the Helm chart(s) being referenced
+    ## nor their chart-specific supported `values`.
+    #
+    # images:
+    #   - appropriate/curl:latest
+    #   - mattermost/mattermost-enterprise-edition:9.9.0
+    images:
+      - name: appropriate/curl
+        newTag: latest
+      - name: mattermost/mattermost-enterprise-edition
+        newTag: 9.9.0
+    actions:
+      onDeploy:
+        after:
+          - description: Validate Mattermost Package
+            maxTotalSeconds: 300
+            wait:
+              cluster:
+                kind: Packages
+                name: mattermost
+                namespace: mattermost
+                condition: "'{.status.phase}'=Ready"
+
+  - name: mattermost-registry1
```

Comment:
how would you build on component vs another?

Reply:
I don't understand. Can you rephrase?

Reply:
I think I answered this below in https://github.com/defenseunicorns/zarf/pull/2690/files/27b8b7d9640e31c00b6f51904e2aba58fe682120#r1679778222

tl;dr: `zarf package create . --include [...<optional component>]`